runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/initializer/SofaRuntimeSpringContextInitializer.java

```diff
@@ -17,23 +17,106 @@
 package com.alipay.sofa.runtime.spring.initializer;
 
 import com.alipay.sofa.common.log.Constants;
+import com.alipay.sofa.infra.constants.SofaBootInfraConstants;
 import com.alipay.sofa.infra.log.space.SofaBootLogSpaceIsolationInit;
 import com.alipay.sofa.infra.utils.SOFABootEnvUtils;
+import com.alipay.sofa.runtime.SofaFramework;
+import com.alipay.sofa.runtime.api.client.ReferenceClient;
+import com.alipay.sofa.runtime.api.client.ServiceClient;
+import com.alipay.sofa.runtime.client.impl.ClientFactoryImpl;
+import com.alipay.sofa.runtime.component.impl.StandardSofaRuntimeManager;
+import com.alipay.sofa.runtime.constants.SofaRuntimeFrameworkConstants;
+import com.alipay.sofa.runtime.service.client.ReferenceClientImpl;
+import com.alipay.sofa.runtime.service.client.ServiceClientImpl;
+import com.alipay.sofa.runtime.service.impl.BindingAdapterFactoryImpl;
+import com.alipay.sofa.runtime.service.impl.BindingConverterFactoryImpl;
+import com.alipay.sofa.runtime.spi.binding.BindingAdapter;
+import com.alipay.sofa.runtime.spi.binding.BindingAdapterFactory;
+import com.alipay.sofa.runtime.spi.client.ClientFactoryInternal;
+import com.alipay.sofa.runtime.spi.component.SofaRuntimeContext;
+import com.alipay.sofa.runtime.spi.component.SofaRuntimeManager;
 import com.alipay.sofa.runtime.spi.log.SofaLogger;
 import com.alipay.sofa.runtime.spi.log.SofaRuntimeLoggerFactory;
+import com.alipay.sofa.runtime.spi.service.BindingConverter;
+import com.alipay.sofa.runtime.spi.service.BindingConverterFactory;
+import com.alipay.sofa.runtime.spring.*;
+import com.alipay.sofa.runtime.spring.bean.SofaParameterNameDiscoverer;
+import com.alipay.sofa.runtime.spring.configuration.SofaRuntimeAutoConfiguration;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory;
 import org.springframework.context.ApplicationContextInitializer;
 import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.core.env.Environment;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import java.util.HashSet;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * @author xuanbei 18/3/13
  */
 public class SofaRuntimeSpringContextInitializer
                                                 implements
                                                 ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+    /**
+     * Objects shared by multi spring application context
+     */
+    private static BindingConverterFactory bindingConverterFactory;
+    private static BindingAdapterFactory   bindingAdapterFactory;
+    private static SofaRuntimeContext      sofaRuntimeContext;
+    private static AtomicBoolean           isInitiated = new AtomicBoolean(false);
+
+    public static void initApplicationContext(ConfigurableApplicationContext applicationContext) {
```

Comment:
Can this class and its method change to not static?

Reply:
it's better to remove the modifier of `public`.  but it can't.